docs/kop.md

```diff
@@ -0,0 +1,135 @@
+---
+dockerfile: "https://hub.docker.com/r/streamnative/kop(TBD)"
+---
+
+KoP (Kafka on Pulsar) brings the native Apache Kafka protocol support to Apache Pulsar by introducing a Kafka protocol handler on Pulsar brokers. By adding the KoP protocol handler to your existing Pulsar cluster, you can migrate your existing Kafka applications and services to Pulsar without modifying the code. This enables Kafka applications to leverage Pulsar’s powerful features, such as:
+
+- Streamlined operations with enterprise-grade multi-tenancy
+- Simplified operations with a rebalance-free architecture
+- Infinite event stream retention with Apache BookKeeper and tiered storage
+- Serverless event processing with Pulsar Functions
+
+KoP, implemented as a Pulsar [protocol handler](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/protocol/ProtocolHandler.java) plugin with the protocol name "kafka", is loaded when Pulsar broker starts. It helps reduce the barriers for people adopting Pulsar to achieve their business success by providing a native Kafka protocol support on Apache Pulsar. By integrating the two popular event streaming ecosystems, KoP unlocks new use cases. You can leverage advantages from each ecosystem and build a truly unified event streaming platform with Apache Pulsar to accelerate the development of real-time applications and services.
+
+KoP implements the Kafka wire protocol on Pulsar by leveraging the existing components (such as topic discovery, the distributed log library - ManagedLedger, cursors and so on) that Pulsar already has.
+
+The following figure illustrates how the Kafka-on-Pulsar protocol handler is implemented within Pulsar.
+
+![](kop-architecture.png)
+
+# Get Started with KoP
+
+If you have an Apache Pulsar cluster, you can enable Kafka-on-Pulsar on your existing Pulsar cluster by downloading and installing the KoP protocol handler to Pulsar brokers directly. It takes three steps:
+1. Download KoP protocol handler, or build the `./kafka-impl/target/pulsar-protocol-handler-kafka-{{connector:version}}.nar` file, and then copy it to your Pulsar `/protocols` directory.
+2. Set the configuration of the KoP protocol handler in Pulsar `broker.conf` or `standalone.conf` files.
+3. Restart Pulsar brokers to load KoP protocol handler.
+
+And then you can start your broker and use KoP. The followings are detailed instructions for each step.
+
+## Download/Build KoP protocol handler
+
+You can download the KoP protocol handler [here](https://github.com/streamnative/kop/releases), or you can build the KoP protocol handler from source code.
+
+1. Clone the KoP GitHub project to your local. 
+
+```bash
+git clone https://github.com/streamnative/kop.git
+cd kop
+```
+
+2. Build the project.
+```bash
+mvn clean install -DskipTests
+```
+
+3. Get the nar file in the following directory and copy it your Pulsar `/protocols` directory.
```

Comment:
before this step, there should be a step to create the Pulsar `/protocols` directory.

Reply:
In addition, it's better to say `protocols` instead of `/protocols` because `/protocols` may be misunderstood to an absolute path. But the directory should be a relative path to pulsar's directory.

Reply:
Thank you.
If it's the first time to protocol handlers, users need to create the `protocols` folder.
If it's not the first time to use it, for example, they have built kop, mop, aop before, they already have created this folder. I'll add description for it.